python/apps/taiga/src/taiga/attachments/apps.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# Copyright (c) 2023-present Kaleidos INC
+
+from taiga.base.django import apps
+
+
+class AttachmentConfig(apps.AppConfig):
+    name = "taiga.attachments"
+    label = "attachments"
+    verbose_name = "Attachments"
+
+    def ready(self) -> None:
+        from taiga.attachments.signals import mark_attachment_file_to_delete
+        from taiga.base.db.models import signals
+
+        signals.post_delete.connect(
```

Comment:
It's probably me but, couldn't we use the ORM's django.db.models.signals.pre_delete signal?

Reply:
This is what [Django documentation](https://docs.djangoproject.com/en/4.2/topics/signals/) say about that:

> ### Where should this code live?
>
> Strictly speaking, signal handling and registration code can live anywhere you like, although it’s recommended to avoid the application’s root module and its models module to minimize side-effects of importing code.
>
> In practice, signal handlers are usually defined in a signals submodule of the application they relate to. Signal receivers are connected in the [ready()](https://docs.djangoproject.com/en/dev/ref/applications/#django.apps.AppConfig.ready) method of your application [configuration class](https://docs.djangoproject.com/en/dev/ref/applications/#configuring-applications-ref). If you’re using the [receiver()](https://docs.djangoproject.com/en/dev/topics/signals/#django.dispatch.receiver) decorator, import the signals submodule inside [ready()](https://docs.djangoproject.com/en/dev/ref/applications/#django.apps.AppConfig.ready), this will implicitly connect signal handlers

I only have a soft preference to follow the official recommendation. Why do you prefer to put this in the models.py?

Reply:
It was just the place I'd look for that opperational logic related to the model, as we do when we overwrite the `save()` method. I was indeed considering avoid usign signals and overwrite the model's `delete()` method, as the [documentation](https://docs.djangoproject.com/en/4.2/ref/signals/#module-django.db.models.signals) suggest.

> Warning
>
> Signals can make your code harder to maintain. Consider implementing a helper method on a [custom manager](https://docs.djangoproject.com/en/4.2/topics/db/managers/#custom-managers), to both update your models and perform additional logic, or else [overriding model methods](https://docs.djangoproject.com/en/4.2/topics/db/models/#overriding-model-methods) before using model signals.

But then I understood why we should be using signals. This method are [not necessarily called](https://docs.djangoproject.com/en/4.2/topics/db/models/#overriding-model-methods)  in a cascaded delete:

> Overridden model methods are not called on bulk operations
>
> Note that the [delete()](https://docs.djangoproject.com/en/4.2/ref/models/instances/#django.db.models.Model.delete) method for an object is not necessarily called when [deleting objects in bulk using a QuerySet](https://docs.djangoproject.com/en/4.2/topics/db/queries/#topics-db-queries-delete) or as a result of a [cascading delete](https://docs.djangoproject.com/en/4.2/ref/models/fields/#django.db.models.ForeignKey.on_delete). To ensure customized delete logic gets executed, you can use [pre_delete](https://docs.djangoproject.com/en/4.2/ref/signals/#django.db.models.signals.pre_delete) and/or [post_delete](https://docs.djangoproject.com/en/4.2/ref/signals/#django.db.models.signals.post_delete) signals.
>
> Unfortunately, there isn’t a workaround when [creating](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#django.db.models.query.QuerySet.bulk_create) or [updating](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#django.db.models.query.QuerySet.update) objects in bulk, since none of [save()](https://docs.djangoproject.com/en/4.2/ref/models/instances/#django.db.models.Model.save), [pre_save](https://docs.djangoproject.com/en/4.2/ref/signals/#django.db.models.signals.pre_save), and [post_save](https://docs.djangoproject.com/en/4.2/ref/signals/#django.db.models.signals.post_save) are called.

So, answering myself, if we are forced to use signal that's the right place to put them (following the recommendations). 

Sorry for the misunderstood.